### DISCUSSION TITLE
Technical limitations of PR reviews in regard to merge conflicts

### DISCUSSION THREAD

**Discussion by flauschzelle** [timhutton/twitter-archive-parser#139] [General]
One thing that stood out to me while working on several PRs in parallel:
Every time one PR is merged, all other PRs get marked as 'can not be merged because there are merge conflicts'...

As long as all the affected PRs are my own, I can at least partially fix that by merging the (then newly updated) current main into my feature branch (again) and pushing the changes.
That makes the 'conflicted' marker disappear - but any 'Approved' reviews get automatically dismissed by that as well and the PR has to be approved by someone else again (because obviously I can't approve my own PRs)... and then I have to be quick to merge it, and repeat the process for the next already-had-been-approved PR, and so on...

And if I see a PR from someone else that was approved by you @timhutton, and I want to help by merging it - I can't because obviously, I can't merge the current main into someone else's feature branch. I think nobody can do that except the person who created the PR. So the only thing I could do is write a comment and tell the original poster to please merge the updated main into their feature branch again.
And even if they do that quickly, it would also have to get reviewed and re-approved (and then probably merged right away) by you @timhutton shortly after that. Because as soon as anything else is merged, the whole process resets again...

It feels impractical and annoying, and I have no idea how to solve this O.o

**Comment by timhutton** [comment]
If it ever becomes a real blocker, and a PR has been approved with no major changes since: make a new PR by starting from their branch and complete that PR then close the original PR. I think we might need to do this for some of the existing PRs that have been stuck for a while.
